hydrolib/core/io/dflowfm/structure/models.py

```diff
@@ -108,27 +108,62 @@ def check_location(cls, values: dict) -> dict:
         filtered_values = {k: v for k, v in values.items() if v is not None}
         structype = filtered_values.get("type", "").lower()
 
-        if not (structype == "compound" or issubclass(cls, (Compound, Dambreak))):
+        if not (structype == "compound" or issubclass(cls, (Compound))):
             # Compound structure does not require a location specification.
+
+            # Backwards compatibility for old-style polylinefile input field (instead of num/x/yCoordinates):
+            polyline_compatible_structures = dict(
+                pump="Pump",
+                dambreak="Dambreak",
+                gate="Gate",
+                weir="Weir",
+                generalstructure="GeneralStructure",
+            )
```

Comment:
Generally speaking, base classes should not know about their subclasses (see SOLID's Open-Closed principle).
We might want to look into inheritance/polymorphism. So that a e.g. a Pump and a Gate implement their own validation.

Reply:
I agree, shall we follow this up under #436?